photon-client/src/main/scala/com/linkedin/photon/ml/util/Utils.scala

```diff
@@ -294,7 +292,7 @@ protected[ml] object Utils {
   @throws(classOf[IllegalArgumentException])
   def getKeyFromMapOrElse[T](map: Map[String, Any], key: String, elseBranch: Either[String, T]): T = {
     map.get(key) match {
-      case Some(x: T) => x
+      case Some(x: Any) => x.asInstanceOf[T] // to avoid type erasure warning
```

Comment:
IMO leaving the warning in place is better here. This code does the same thing as before, but without any notification that it might behave unexpectedly. Maybe we could include a class tag so that the runtime can resolve the type? Also, use of `asInstanceOf` is generally frowned upon.